src/shared/utilities/vsCodeUtils.ts

```diff
@@ -72,3 +73,51 @@ export async function closeAllEditors() {
         )
     }
 }
+
+/**
+ * Checks if the given extension is installed and active.
+ */
+export function isExtensionActive(extId: string) {
+    const extension = vscode.extensions.getExtension(extId)
+    return extension && !extension.isActive
+}
+
+/**
+ * Activates the given extension, or does nothing if the extension is not
+ * installed.
+ *
+ * @param extId  Extension id
+ * @param silent  Return undefined on failure, instead of throwing
+ * @returns Extension object, or undefined on failure if `silent`
+ */
+export async function activateExtension(extId: string, silent: boolean = true): Promise<vscode.Extension<void> | undefined> {
+    let log: (s: string, ...rest: any[]) => void
+    try {
+        log = getLogger().debug
+    } catch {
+        log = console.log
+    }
+
+    const extension = vscode.extensions.getExtension(extId)
+    if (!extension) {
+        if (silent) {
+            return undefined
+        }
+        throw new Error(`Extension not found: ${extId}`)
+    }
+
+    if (!extension.isActive) {
+        try {
+            await extension.activate()
+            log('Extension activated: %s', extId)
+        } catch (err) {
+            if (!silent) {
+                throw err
+            }
+            log('Extension failed to activate: %s: %O', extId, err as Error)
```

Comment:
Log before throw

Reply:
Intentionally did not. Because the exception will be logged by higher layers.

Reply:
But the "Extension failed to activate:" semantics is useful to have. I will make a change...